plugins/pom-analyzer/src/main/java/eu/f4sten/pomanalyzer/utils/DatabaseUtils.java

```diff
@@ -52,28 +48,18 @@ protected MetadataDao getDao(DSLContext ctx) {
     }
 
     public void save(Pom result) {
-        var hasProcessedRecord = new AtomicBoolean(false);
-        int numTries = 0;
-        while (!hasProcessedRecord.get() && numTries < Constants.transactionRestartLimit) {
-            numTries++;
+        try {
             context.transaction(transaction -> {
                 var dao = getDao(DSL.using(transaction));
-                try {
-                    insertIntoDB(result, dao);
-                    hasProcessedRecord.set(true);
-                } catch (DataAccessException e) {
-                    // Can be happen. Normally fixable through retrying.
-                    LOG.debug("Caught DataAccessException during insertion of Pom. Retrying ...");
-                }
+                insertIntoDB(result, dao);
             });
-        }
-        if (!hasProcessedRecord.get()) {
-            // catch cases in which retries did not work
-            throw new IllegalStateException("Unable to save Pom in database");
+        } catch (DataAccessException e) {
+            throw new UnrecoverableError(e);
         }
     }
 
-    public void insertIntoDB(Pom r, MetadataDao dao) {
+    @SuppressWarnings("deprecation")
```

Comment:
Why is this method deprecated now?

Reply:
People should use `safe`, which also includes the error-checking logic...